### ISSUE TITLE
General project management - komunigi - 2 - technical axis

### ISSUE THREAD

**Issue by robinkeunen** [coopiteasy/vertical-cooperative#204]
As of early 2022, axis 2 of the **komunigi** project is starting in earnest.

For context, see the overarching issues [RFC Moving Obeesdoo and Easy My Coop to OCA #247](https://github.com/beescoop/Obeesdoo/issues/247) and [RFC Moving Easy My Coop to OCA #170](https://github.com/coopiteasy/vertical-cooperative/issues/170).

The present issue encompasses project management tasks relating to the technical axis (2) of the komunigi project, ie to the two issues above.

Moving Obeesdoo and EMC to the OCA will require some project management tasks, notably :

- Organising and following the work module by module
- Organising regular sprints and follow-up
- Communicating about changes to users and publish regular release notes
- Maintaining and updating a public window to our progress within the technical axis

Moved from https://github.com/beescoop/Obeesdoo/issues/262

Here are the 4 steps of the project as of 2022-02-01

# 1. Rename modules 

cf [Analysis - rename repositories and module #248](https://github.com/beescoop/Obeesdoo/issues/248)

## 1.1 vertical-cooperative

**Depth 1**
- https://github.com/coopiteasy/vertical-cooperative/issues/203 : 32h (+ 8h already done)

**Depth 2**
- https://github.com/coopiteasy/vertical-cooperative/issues/207 : 8h
- https://github.com/coopiteasy/vertical-cooperative/issues/208 : 8h
- https://github.com/coopiteasy/vertical-cooperative/issues/209 : 8h
- https://github.com/coopiteasy/vertical-cooperative/issues/210 : 8h
- https://github.com/coopiteasy/vertical-cooperative/issues/211 : 16h
- https://github.com/coopiteasy/vertical-cooperative/issues/213 : 16h
- [x] `easy_my_coop_api_logs` : 16h
- https://github.com/coopiteasy/vertical-cooperative/issues/214 : 32h
- https://github.com/coopiteasy/vertical-cooperative/issues/215 : 16h
- https://github.com/coopiteasy/vertical-cooperative/issues/216 : 16h
- https://github.com/coopiteasy/vertical-cooperative/issues/217 : 16h
- `beesdoo_easy_my_coop` : 16h => rather do it as part of obeesdoo

**Depth 3**
- ~~`easy_my_coop_connector` : 32h~~
- https://github.com/coopiteasy/vertical-cooperative/issues/216 : 8h
- ~~`easy_my_coop_loan_website`~~ : 8h => leave loans on the side at this point : seldom used and not part of cooperatives
- ~~`easy_my_coop_loan_account`~~ : 8h
- ~~`easy_my_coop_loan_bba`~~ : 8h

**Depth 4**
- ~~`easy_my_coop_loan_account_be`~~ : 8h

![](https://user-images.githubusercontent.com/17589077/149750369-34ea7875-4afa-41dc-afa9-cb66113ba6bc.png)

## 1.2 Obeesdoo

![](https://user-images.githubusercontent.com/17589077/149927982-58f1b329-9d15-48d3-b331-77741f7c2e87.png)
cf https://github.com/beescoop/Obeesdoo/issues/248

# 2. rename models and xml_id suffixes

- remove mentions to beesdoo and easy_my_coop
- remove mentions to cooperative in shift modules (ex `cooperative.status` -> `worker.status`)
- give a second think to other model names.

# 3. Technical refactoring 

- refacto already listed in issues
- todo and fixme scattered around the code (convert to issues as we go)

# 4. UX refactoring

- change of menu hierarchy
- change in form / tree views
- move configurations to better places
- analyse with users

**Comment by robinkeunen** [comment]
Done on 2022-02-11 :

- creation of 
    - https://github.com/oca/cooperative
    - https://github.com/oca/shift-planning
    - https://github.com/oca/vertical-cooperative-supermarket
- backport (sort of) OCA-CI on these repositories on branch 12.0
- creation of cooperative PSC (Robin, Rémy, Houssine, Eugeni)
- renaming of
    - easy_my_coop, easy_my_coop_be, easy_my_coop_fr, easy_my_coop_api
    - use partner_age from the oca rather than custom_module
    - migration script
- integration of Polln contributions (30+ PR by Thibault)

**Comment by robinkeunen** [comment]
2022-04-28 @victor-champonnois @carmenbianca @polchampion 

# State

- Renaming `easy_my_coop` modules is done [emc project](https://github.com/orgs/coopiteasy/projects/2/views/2)
- Unbranding `cooperator` modules is mostly done [emc project](https://github.com/orgs/coopiteasy/projects/2/views/2)
- Extracting beesdoo_x functionalities to new modules has started [beesdoo project](https://github.com/orgs/beescoop/projects/2/views/2) 
	- https://github.com/beescoop/Obeesdoo/issues/333
	- https://github.com/beescoop/Obeesdoo/issues/332

# Next Steps EMC

Coop IT Easy's database will be migrated from 9.0 to 12.0 during the month of May.

1a) fix last test feedbacks [issue link](xxx) and deploy on komunigi server
	- https://github.com/coopiteasy/vertical-cooperative/issues/214
	- https://github.com/coopiteasy/vertical-cooperative/issues/221
	- https://github.com/coopiteasy/vertical-cooperative/issues/295
1b) merge all _unbranding_ PR and deploy on Komunigi server
2) once fixed, deploy branch 12.0-komunigi on mutual test server by mid-may.

The goal is to merge 12.0-komunigi into 12.0 and deploy on all databases in production in early June. We want to have 12.0-komunigi deployed on our test server for two weeks before deploying in production.

# Next Steps beesdoo

Keep extracting functionalities from beesdoo_x (cf task list). These developments should be deployed directly on the mutualised test server

Before leaving, I'll analyse further how to split `beesdoo_product`

## Notes from meeting with @robinkeunen and @victor-champonnois  

The main idea is to split the dependency tree in several independent trees, by splitting the root module beesdoo_base in different root modules. See https://github.com/beescoop/Obeesdoo/issues/247 on how to do that.

### about migration scripts 
- if a field, is removed from a model, is it removed in the DB ?
- if yes, needs to do a two step migrations:
    - pre-migration : create a backup of the column 
    - post-migration : move this backup to the table of the new module
- what is the best way to rename the module, when splitting module A into B and C ?
    - create B and rename A into C ?
    - or create B and C and remove A ?

**Comment by victor-champonnois** [comment]
@robinkeunen 
Recap vertical-cooperative:
- tests are OK (see [here](https://docs.google.com/spreadsheets/d/11mkW2B-j0T1avBgMMcGClrWtRO5OUlbUUv5XoMdd1fY/edit#gid=791555568))
- deployed in 12-test since 1st of June
- need to review this : https://github.com/coopiteasy/vertical-cooperative/issues/219
- See the discussion with Hugues about integrating migration script in modules https://github.com/coopiteasy/vertical-cooperative/issues/233. (in short, there seems to be a working solution but we don't have time to implement it for vertical-cooperative, we'll try it on obeesdoo)

**Comment by victor-champonnois** [comment]
Recap Obeesdoo:
- Beesdoo_account refactoring is deployed, tested and OK. https://github.com/beescoop/Obeesdoo/pull/385
- Beesdoo_base refactoring is mostly done, with some interrogation about what to do with remaining functionalities. https://github.com/beescoop/Obeesdoo/pull/384
- Beesdoo_product and beesdoo_website_eater : started to work on it 
- We should try to have all migration scripts inside modules (see above)

**Comment by robinkeunen** [comment]
2022-07-28 state of komunigi

2022-04-28 @victor-champonnois @carmenbianca @polchampion 

# EMC 
## Done

- Renaming `easy_my_coop` modules is done [emc project](https://github.com/orgs/coopiteasy/projects/2/views/2)
- Unbranding `cooperator` modules is done [emc project](https://github.com/orgs/coopiteasy/projects/2/views/2)
- All of this was deployed for our v12 customers

## Next Steps

- https://github.com/coopiteasy/vertical-cooperative/issues/359
	- depending on :
	- https://github.com/coopiteasy/vertical-cooperative/issues/357
	- https://github.com/coopiteasy/vertical-cooperative/issues/350
	- https://github.com/coopiteasy/vertical-cooperative/issues/227
	- https://github.com/coopiteasy/vertical-cooperative/issues/356
        	- https://github.com/coopiteasy/vertical-cooperative/pull/349

# Obeesdoo

This code is still at the stage of renaming and splitting the modules. 
There is still much work to do. The developpement is slowed by recurring git conflicts with ongoing development.
We need tests from the analysts to merge the "in progress" PRs.

## Done

- https://github.com/beescoop/Obeesdoo/pull/413
	- https://github.com/beescoop/Obeesdoo/issues/332
	- https://github.com/beescoop/Obeesdoo/issues/333
- https://github.com/beescoop/Obeesdoo/pull/420
	- https://github.com/beescoop/Obeesdoo/issues/348
	- https://github.com/beescoop/Obeesdoo/issues/349
	- https://github.com/beescoop/Obeesdoo/issues/350
	- https://github.com/OCA/stock-logistics-workflow/pull/1026

## In Progress

- https://github.com/beescoop/Obeesdoo/pull/384
	- https://github.com/beescoop/Obeesdoo/issues/335
	- https://github.com/beescoop/Obeesdoo/issues/334
	- https://github.com/beescoop/Obeesdoo/issues/336
- https://github.com/beescoop/Obeesdoo/pull/411
    - https://github.com/beescoop/Obeesdoo/issues/392
    - https://github.com/beescoop/Obeesdoo/issues/409
    - https://github.com/beescoop/Obeesdoo/issues/391
- https://github.com/beescoop/Obeesdoo/pull/427
   - https://github.com/beescoop/Obeesdoo/issues/337
   - https://github.com/beescoop/Obeesdoo/issues/276
   - https://github.com/beescoop/Obeesdoo/issues/338
   - https://github.com/beescoop/Obeesdoo/issues/339
- https://github.com/beescoop/Obeesdoo/pull/387
	- https://github.com/beescoop/Obeesdoo/issues/341

## Next Steps

Merge these PR one by one.

cf https://github.com/orgs/beescoop/projects/2/views/1?filterQuery=status%3ATodo

**Comment by robinkeunen** [comment]
2022-09-26

# vertical-cooperative

- renamed models and xml_id suffixes
- blocked by mig14, ok now
- starting to move to oca/cooperative by the end of the week

- 10 refactoring issues in the backlog
- going through them when stuck on obeesdoo

| Title | Priorities | Time |
|---|---|---|
| UX: reorganise configurations https://github.com/OCA/cooperative/issues/26|  1 | 16h |
| ~~Remove disabled constraint unique_number on res.partner.bank~~ #241 | 1 |  4h |
| Remove duplication from email templates https://github.com/OCA/cooperative/issues/14 |  2 | 16h |
| Refactor states in easy_my_coop https://github.com/OCA/cooperative/issues/28 |  2 | 24h |
| Rename subscription.request to coop.share.request https://github.com/OCA/cooperative/issues/20 | 3 |  8h |
| Incorrect computation of cooperator_type ? https://github.com/OCA/cooperative/issues/18 |  3 | 40h |
| Refactor `subscription/subscribe_share ` route so that noqa: C901 (method too complex) can be removed https://github.com/coopiteasy/vertical-cooperative/issues/303 | 4 |  8h |
| Remove `sale` from cooperator dependencies? https://github.com/OCA/cooperative/issues/16 | 4 |  4h |
| 12.0 move send_mail_with_multiple_attachments in separate module https://github.com/OCA/cooperative/issues/17 | 4 |  8h |
| cooperator: rename old_member to former_member https://github.com/OCA/cooperative/issues/15| 5 |  4h |
| **Total** | **116h + xh** | |


# Obeesdoo

This code is still at the stage of renaming and splitting the modules. 
We need tests from the analysts to merge the "in progress" PRs.

## Done

- https://github.com/beescoop/Obeesdoo/pull/413
	- https://github.com/beescoop/Obeesdoo/issues/332
	- https://github.com/beescoop/Obeesdoo/issues/333
- https://github.com/beescoop/Obeesdoo/pull/420
	- https://github.com/beescoop/Obeesdoo/issues/348
	- https://github.com/beescoop/Obeesdoo/issues/349
	- https://github.com/beescoop/Obeesdoo/issues/350
	- https://github.com/OCA/stock-logistics-workflow/pull/1026
- https://github.com/beescoop/Obeesdoo/pull/384
	- https://github.com/beescoop/Obeesdoo/issues/335
	- https://github.com/beescoop/Obeesdoo/issues/334
	- https://github.com/beescoop/Obeesdoo/issues/336
	
## Waiting for tests

- https://github.com/beescoop/Obeesdoo/pull/411
    - https://github.com/beescoop/Obeesdoo/issues/392
    - https://github.com/beescoop/Obeesdoo/issues/409
    - https://github.com/beescoop/Obeesdoo/issues/391
- https://github.com/beescoop/Obeesdoo/pull/427
   - https://github.com/beescoop/Obeesdoo/issues/337
   - https://github.com/beescoop/Obeesdoo/issues/276
   - https://github.com/beescoop/Obeesdoo/issues/338
   - https://github.com/beescoop/Obeesdoo/issues/339

## Next Steps

- Split `beesdoo_product` https://github.com/beescoop/Obeesdoo/pull/470 : 44h
    - 8h https://github.com/beescoop/Obeesdoo/issues/401
    - 4h https://github.com/beescoop/Obeesdoo/issues/402
    - 8h https://github.com/beescoop/Obeesdoo/issues/403
    - 4h https://github.com/beescoop/Obeesdoo/issues/404
    - 4h https://github.com/beescoop/Obeesdoo/issues/405
    - 8h https://github.com/beescoop/Obeesdoo/issues/406
    - 8h https://github.com/beescoop/Obeesdoo/issues/408
- Split `beesdoo_pos` https://github.com/beescoop/Obeesdoo/pull/466 : 24h
    - 8h https://github.com/beescoop/Obeesdoo/issues/344
    - 16h https://github.com/beescoop/Obeesdoo/issues/345
- Split `beesdoo_purchase` : 28h
    - 8h https://github.com/beescoop/Obeesdoo/issues/358
    - 4h https://github.com/beescoop/Obeesdoo/issues/356
    - 8h https://github.com/beescoop/Obeesdoo/issues/355
    - 8h https://github.com/beescoop/Obeesdoo/issues/357
- 8h https://github.com/beescoop/Obeesdoo/issues/407
- 32h move `shift` to the OCA

Total : 120h

# Time matters - axe 2

- 969h / 1540h coded out of bugdeted
- in the CP (not updated by Robin): 58 + 120 + 84 = 262h
- 120h of obeesdoo prioitary
- refactoring à prioriser

**Comment by robinkeunen** [comment]
@victor-champonnois Sur base de mes discussions avec @huguesdk , j'ai ajouté une colonne "priorité" au tableau des refactoring ci-dessus.

**Comment by robinkeunen** [comment]
2022-11-22

# oca/cooperative

| module | PR | time |
| :----- | :- | :---|
| portal_recaptcha | | 4h |
| cooperator_website_recaptcha | | 4h |
| l10n_be_cooperator_portal |  | 8h |
| l10n_be_cooperator|  | 4h |
| l10n_ch_cooperator | | 4h |
| l10n_es_cooperator | do not do | - |
| l10n_fr_cooperator | | 4h |
| https://github.com/OCA/cooperative/issues/18 | | 8h+ |
| https://github.com/OCA/cooperative/issues/19 | | 4h+ |
| **Total** | | **28h** | 

# obeesdoo

| module | PR | time |
| :----- | :- | :---|
| `beesdoo_account` | https://github.com/beescoop/Obeesdoo/pull/413 | - | 
| `beesdoo_base` | https://github.com/beescoop/Obeesdoo/pull/384 | - | 
| `beesdoo_crelan_csv` | | 4h | 
| `beesdoo_easy_my_coop` | https://github.com/beescoop/Obeesdoo/pull/427 | - | 
| `beesdoo_inventory` | https://github.com/beescoop/Obeesdoo/pull/420 | - | 
| `beesdoo_pos` | https://github.com/beescoop/Obeesdoo/pull/466 | - | 
| `beesdoo_pos_coop_status` | https://github.com/beescoop/Obeesdoo/pull/467 | - | 
| `beesdoo_pos_email_ticket` | | 4h | 
| `beesdoo_pos_reporting` | https://github.com/beescoop/Obeesdoo/pull/468 | - | 
| `beesdoo_print_label` |  do not do | - | 
| `beesdoo_product` | https://github.com/beescoop/Obeesdoo/pull/470 | 28h | 
| `beesdoo_product_info_screen` | | 4h | 
| `beesdoo_product_usability` | https://github.com/beescoop/Obeesdoo/pull/469 | - | 
| `beesdoo_purchase` | | 32h | 
| `beesdoo_shift` | | 16h | 
| `beesdoo_shift_attendance` | | 4h | 
| `beesdoo_shift_swap` | | 4h | 
| `beesdoo_shift_welcome_screen` |  | 4h | 
| `beesdoo_stock` | https://github.com/beescoop/Obeesdoo/pull/464 | - | 
| `beesdoo_stock_coverage` | | 4h | 
| `beesdoo_website_eater` | https://github.com/beescoop/Obeesdoo/pull/387 | - | 
| `beesdoo_website_posorder_amount` | | 4h | 
| `beesdoo_website_shift` | | 4h | 
| `beesdoo_website_shift_swap` | | 4h | 
| `beesdoo_website_theme` | | 4h | 
| `beesdoo_worker_status` | | 4h | 
| `beesdoo_worker_status_shift_swap` | | 4h | 
| **Total** | | **128h** | 

Total : 156h w/o moving beesdoo modules to the OCA

**Comment by victor-champonnois** [comment]
@robinkeunen beesdoo_print_label will maybe take more than 4 hours, cf our previous discussion : https://github.com/beescoop/Obeesdoo/issues/400

**Comment by victor-champonnois** [comment]
I don't see beesdoo_worker_status in https://user-images.githubusercontent.com/17589077/149927982-58f1b329-9d15-48d3-b331-77741f7c2e87.png, am I missing sthg ?

**Comment by victor-champonnois** [comment]
If we want to be safer, 32h would be better for beesdoo_purchase

**Comment by robinkeunen** [comment]
2023-01-10

# oca/cooperative

| module | PR | comment | time |
| :----- | :- | :------ | :---|
| portal_recaptcha | https://github.com/OCA/website/pull/914 | failing travis|  2h |
| cooperator_website_recaptcha | https://github.com/OCA/cooperative/pull/43 | depends on portal_recaptcha | 2h |
| l10n_be_cooperator_portal | | | 8h |
| l10n_ch_cooperator | | | 4h |
| l10n_es_cooperator | | coopdevs | 0h |
| l10n_fr_cooperator | | | 4h |
| Allow multiple shares and multiple share types | https://github.com/OCA/cooperative/issues/18 | cancel | 0 |
| Permission error validating a draft request | https://github.com/OCA/cooperative/issues/19 | cancel | 0 |
| **Total** | | | **20h** | 

# obeesdoo

| module | PR | comment | time |
| :----- | :- | :------ | :---|
| `beesdoo_pos_email_ticket` | | replace w/ oca pos_invoice_send_mail in 16.0 | 0h | 
| `beesdoo_print_label` | | do not do | - | 
| `beesdoo_product` | https://github.com/beescoop/Obeesdoo/pull/470 | 2 tests + translations | 8 h | 
| `beesdoo_product_info_screen` | https://github.com/beescoop/Obeesdoo/pull/494 | needs review | 2h | 
| `beesdoo_purchase` | https://github.com/beescoop/Obeesdoo/pull/486 | review followup and suggested price refactor | 4h | 
| `beesdoo_shift` | https://github.com/beescoop/Obeesdoo/pull/490 | needs review | 16h | 
| `beesdoo_shift_attendance` | https://github.com/beescoop/Obeesdoo/pull/490 | cf beesdoo_shift | 0h | 
| `beesdoo_shift_swap` | | | 4h | 
| `beesdoo_shift_welcome_screen` | |  | 4h | 
| `beesdoo_stock_coverage` | | | 4h | 
| `beesdoo_website_posorder_amount` | | | 4h | 
| `beesdoo_website_shift` | | | 4h | 
| `beesdoo_website_shift_swap` | | | 4h | 
| `beesdoo_website_theme` | | | 4h | 
| `beesdoo_worker_status` | https://github.com/beescoop/Obeesdoo/pull/490 | cf beesdoo_shift | 0h |
| `beesdoo_worker_status_shift_swap` | | | 4h | 
| **Total** | | | **62h** | 

Total : 82h

**Comment by robinkeunen** [comment]
2023-02-20

# oca/cooperative

| module | PR | comment | time |
| :----- | :- | :------ | :---|
| portal_recaptcha | https://github.com/OCA/website/pull/914 | failing travis|  2h |
| cooperator_website_recaptcha | https://github.com/OCA/cooperative/pull/43 | depends on portal_recaptcha | 2h |
| l10n_be_cooperator_portal | | | 8h |
| l10n_ch_cooperator | | | 4h |
| l10n_es_cooperator | | coopdevs | 0h |
| l10n_fr_cooperator | | | 4h |
| Allow multiple shares and multiple share types | https://github.com/OCA/cooperative/issues/18 | cancel | 0 |
| Permission error validating a draft request | https://github.com/OCA/cooperative/issues/19 | cancel | 0 |
| **Total** | | | **20h** | 

# obeesdoo

| module | PR | comment | time |
| :----- | :- | :------ | :---|
| `beesdoo_pos_email_ticket` | | replace w/ oca pos_invoice_send_mail in 16.0 | 0h | 
| `beesdoo_print_label` | | do not do | - | 
| `beesdoo_product_info_screen` | https://github.com/beescoop/Obeesdoo/pull/494 | needs review | 2h | 
| `beesdoo_purchase` | https://github.com/beescoop/Obeesdoo/pull/486 | to test | 4h | 
| `beesdoo_shift` | https://github.com/beescoop/Obeesdoo/pull/490 | needs review | 16h | 
| `beesdoo_shift_attendance` | https://github.com/beescoop/Obeesdoo/pull/490 | cf beesdoo_shift | 0h | 
| `beesdoo_shift_swap` | | | 4h | 
| `beesdoo_shift_welcome_screen` | |  | 4h | 
| `beesdoo_stock_coverage` | https://github.com/coopiteasy/cie-stock/pull/3| | 4h | 
| `beesdoo_website_posorder_amount` | | | 4h | 
| `beesdoo_website_shift` | | | 4h | 
| `beesdoo_website_shift_swap` | | | 4h | 
| `beesdoo_website_theme` | | | 4h | 
| `beesdoo_worker_status` | https://github.com/beescoop/Obeesdoo/pull/490 | cf beesdoo_shift | 0h |
| `beesdoo_worker_status_shift_swap` | | | 4h | 
| **Total** | | | **54h** | 

Total : 74h
